tests/binary/run.py

```diff
@@ -1,131 +1,107 @@
-
-  for v in [("0", False), ("1000000011110000", True),
-              ("111111110000000011111111", False)]:
-    try:
-      t.container.b3 = v[0]
-      passed = True
-    except ValueError:
-      passed = False
-    assert passed == v[1], \
-        "limited length binary incorrectly set to %s (%s != %s)" \
-          % (v[0], v[1], passed)
-
-  for v in [("0", False), ("1111111100000000", True),
-            ("111111110000000011111111", True),
-            ("1111111100000000111111110000000011110000", False)]:
-    try:
-      t.container.b4 = v[0]
-      passed = True
-    except ValueError:
-      passed = False
-    assert passed == v[1], "limited length binary with range incorrectly " + \
-      "set to %s (%s != %s)" % (v[0], v[1], passed)
-
-  for v in [("0", False), ("1111000011110000", True),
-            ("111100001111000011110000", True),
-            ("1111000011110000111100001111000011110000", False),
-            ("111100001111000011110000111100001111000011110000", True),
-            ("111100001111000011110000111100001111000011110000" +
-              "11110000111100001111000011110000", True),
-            ("111100001111000011110000111100001111000011110000" +
-              "111100001111000011110000111100001111000011110000" +
-              "111100001111000011110000111100001111000011110000" +
-              "111100001111000011110000111100001111000011110000" +
-              "111100001111000011110000111100001111000011110000" +
-              "111100001111000011110000111100001111000011110000" +
-              "111100001111000011110000111100001111000011110000" +
-              "1010101010101010", False)]:
-    try:
-      t.container.b5 = v[0]
-      passed = True
-    except Exception:
-      passed = False
-    assert passed == v[1], "limited length binary with complex length " + \
-        "argument incorrectly set to %s (%s != %s)" % (v[0], v[1], passed)
-
-  if not k:
-    os.system("/bin/rm %s/bindings.py" % this_dir)
-    os.system("/bin/rm %s/bindings.pyc" % this_dir)
+import unittest
+
+from tests.base import PyangBindTestCase
+
+
+class BinaryTests(PyangBindTestCase):
+  yang_files = ['binary.yang']
+
+  def setUp(self):
+    self.binary_obj = self.bindings.binary()
+
+  def test_binary_leafs_exist(self):
+    for leaf in ["b1", "b2", "b3"]:
+      with self.subTest(leaf=leaf):
+        self.assertTrue(hasattr(self.binary_obj.container, leaf), "Element did not exist in container (%s)" % leaf)
+
+  def test_set_bitarray_from_different_datatypes(self):
+    for value in [("01110", True), ({"42": 42}, True), (-42, False), ("Arthur Dent", False)]:
+      with self.subTest(value=value):
+        allowed = True
+        try:
+          self.binary_obj.container.b1 = value[0]
+        except ValueError:
+          allowed = False
+        self.assertEqual(allowed, value[1],
+          "Could incorrectly set b1 to %s" % value[0])
+
+  def test_binary_leaf_default_value(self):
+    default = self.bindings.bitarray("0100")
```

Comment:
What's the logic to call `bitarray` from the `self.bindings` class, rather than simply import it here? (`from bitarray import bitarray`?)

Reply:
Absolutely none! :D Total oversight on my part. I'll fix that up.